src/submission/submission-service.ts

```diff
@@ -892,25 +896,59 @@ export namespace operations {
           .processParallel(command.clinicalType, record, index, schema);
 
         if (schemaResult.validationErrors.length > 0) {
-          let validationErrors = schemaResult.validationErrors;
+          let validationErrors: DeepReadonly<SchemaValidationError>[] = [
+            ...schemaResult.validationErrors,
+          ];
+
           if (FEATURE_SUBMISSION_EXCEPTIONS_ENABLED) {
+            // check for program exception
             const result = await programExceptionRepository.find(command.programId);
 
-            // filter out valid exceptions before adding to error accumulator
-            validationErrors =
-              // only filter is we have valid exceptions available
-              isProgramException(result)
-                ? schemaResult.validationErrors.filter(validationError => {
-                    return !applyExceptionIfExists(result.exceptions, validationError, record);
-                  })
-                : schemaResult.validationErrors;
+            validationErrors = [];
+
+            if (isProgramException(result)) {
+              schemaResult.validationErrors.forEach(validationError => {
+                const validationErrorFieldName = validationError.fieldName;
+                const recordValue = record[validationErrorFieldName];
+
+                // Zero Array type exceptions exist, but recordValue type is string | string[]
+                if (Array.isArray(recordValue)) {
+                  validationErrors.push(validationError);
+                  return;
+                }
+
+                const normalizedRecordValue = normalizeExceptionValue(recordValue);
+
+                const exceptionExists = checkExceptionExists(
+                  result.exceptions,
+                  validationError,
+                  normalizedRecordValue,
+                );
+
+                if (exceptionExists) {
+                  // ensure value is normalized exception value
+                  const normalizedExceptionRecord = {
+                    ...record,
+                    ...{ [validationErrorFieldName]: normalizedRecordValue },
```

Comment:
spread + object = don't need either
just put `[validationErrorFieldName]: normalizedRecordValue`

Reply:
you mean without '...{', right?